### PR TITLE
Skip tests for CI with SDC_CONFIG_PIPELINE_SDC=False

### PR DESCRIPTION
**Pull request by PokhodenkoSA**
This PR is based on PR from @Vyacheslav-Smirnov #301 which configures CI for running with SDC_CONFIG_PIPELINE_SDC=False.

This PR introduces convenient decorators `skip_numba_jit` and `skip_sdc_jit` for skipping tests and easy search it in code.
Inside this decorators implementations you can see in code how to switch them off (for unskip tests during development) by uncommenting one line.

I have based on the commit which is before #338 merged. From discussion I understand that it is not ready yet to be merged. If it is ok with this merge then my PR should be rebased.

This PR removes TestSeriesNumba because I understand that running numba.jit with SDC_CONFIG_PIPELINE_SDC=True has no sense. So number of tests for series is not doubled again.

In conclusion, my and @Vyacheslav-Smirnov modifications both enable CI with SDC_CONFIG_PIPELINE_SDC=False and skips tests for making it green. I hope so.

### CODE REVIEW

File: sdc/tests/test_utils.py [hunk 1]
```diff
@@ -108,3 +110,55 @@ def get_start_end(n):
 
 def check_numba_version(version):
     return numba.__version__ == version
+
+
+def msg_and_func(msg_or_func=None):
+    if msg_or_func is None:
+        # No signature, no function
```
Comment (PokhodenkoSA) [inline sdc/tests/test_utils.py]:
```suggestion
        # No message, no function
```

**Comment by shssf** [comment]
> From discussion I understand that it is not ready yet to be merged

I know nothing about it. Please ask to merge this PR for people with whom you discussed it

**Comment by PokhodenkoSA** [comment]
> > From discussion I understand that it is not ready yet to be merged
> 
> I know nothing about it. Please ask to merge this PR for people with whom you discussed it
I mean #338 was not ready to be merged - it is PR from @Hardcode84 with porting to master. Us I understand this PR was not ready to be merged in master. @Hardcode84, is it correct?

So I think the merge of PR #338 to master will be reverted. So for my PR I take a commit before the merge commit.

**Comment by PokhodenkoSA** [comment]
Now #338 is reverted by #342. I have rebased on master and merged #301 for activating new pipeline.
